spk/dnscrypt-proxy/src/service-setup.sh

```diff
@@ -8,34 +8,72 @@ BACKUP_PORT="10053"
+# DSM 6.2 example: DSM Version: 6.2-23739
+UNAME=$(uname -a)
+echo "uname: $UNAME" >> "${INST_LOG}" 2>&1
+# SRM example: Linux {some-name} 4.4.60 #7779 SMP Mon Jan 28 04:30:39 CST 2019 armv7l GNU/Linux synology_ipq806x_rt2600ac
+# DSM example: Linux {some-name} 3.10.105 #23739 SMP Tue Jul 3 19:47:13 CST 2018 x86_64 GNU/Linux synology_bromolow_3615xs
+OS="dsm"
+if echo "$UNAME" | grep -q -i 'rt1900ac\|rt2600ac\|mr2200ac'; then
+    OS="srm"
+fi
+echo "OS detected: $OS" >> "${INST_LOG}" 2>&1
+
 blocklist_setup () {
     ## https://github.com/jedisct1/dnscrypt-proxy/wiki/Public-blacklists
     ## https://github.com/jedisct1/dnscrypt-proxy/tree/master/utils/generate-domains-blacklists
     echo "Install/Upgrade generate-domains-blacklist.py (requires python)" >> "${INST_LOG}"
     mkdir -p "${SYNOPKG_PKGDEST}/var"
-    touch "${SYNOPKG_PKGDEST}"/var/ip-blacklist.txt
+    touch "${SYNOPKG_PKGDEST}"/var/ip-blocklist.txt
     if [ ! -e "${SYNOPKG_PKGDEST}/var/domains-blacklist.conf" ]; then
         wget -t 3 -O "${SYNOPKG_PKGDEST}/var/domains-blacklist.conf" \
             --https-only https://raw.githubusercontent.com/jedisct1/dnscrypt-proxy/master/utils/generate-domains-blacklists/domains-blacklist.conf
     fi
 }
 
+blocklist_cron_uninstall () {
+    # remove cron job
+    sed -i '/.*update-blocklist.sh/d' /etc/crontab
+    synoservicectl --restart crond >> "${INST_LOG}" 2>&1
+}
+
 pgrep () {
-    # shellcheck disable=SC2009,SC2153
-    ps -w | grep "[^]]$1" >> "${LOG_FILE}" 2>&1
+    if [ "$OS" == 'dsm' ]; then
+        # shellcheck disable=SC2009,SC2153
+        ps aux | grep "$1" >> "${LOG_FILE}" 2>&1
+    else
+        # shellcheck disable=SC2009,SC2153
+        ps -w | grep "[^]]$1" >> "${LOG_FILE}" 2>&1
+    fi
+}
+
+restart_dhcpd () {
+    /etc/rc.network nat-restart-dhcp >> "${LOG_FILE}" 2>&1
 }
 
 forward_dns_dhcpd () {
+    echo "dns forwarding - $1" >> "${LOG_FILE}"
     if [ "$1" == "no" ] && [ -f /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.conf ]; then
-        echo "dns forwarding - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
-        echo "enable=\"$1\"" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.info
-        /etc/rc.network nat-restart-dhcp >> "${LOG_FILE}" 2>&1
+        if [ "$OS" == 'dsm' ]; then
+            echo "enable=no" > /etc/dhcpd/dhcpd-dns-dns.info
+        else
+            echo "enable=no" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.info
+        fi
+        restart_dhcpd
     elif [ "$1" == "yes" ]; then
         if pgrep "dhcpd.conf"; then  # if dhcpd (dnsmasq) is enabled and running
-            echo "dns forwarding - dhcpd (dnsmasq) enabled: $1" >> "${LOG_FILE}"
-            echo "server=127.0.0.1#${BACKUP_PORT}" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.conf
-            echo "enable=\"$1\"" > /etc/dhcpd/dhcpd-dnscrypt-dnscrypt.info
-            /etc/rc.network nat-restart-dhcp >> "${LOG_FILE}" 2>&1
+            if [ "$OS" == 'dsm ' ]; then
+                echo "server=127.0.0.1#${BACKUP_PORT}" > /etc/dhcpd/dhcpd-dns-dns.conf
```

Comment:
I wonder why configuration file name may not be the same between DSM 5.2 (SRM) and DSM 6... Maybe because of firewall "sc" file where title is "DNS"?

Reply:
So the SRM and DSM have a different script for combining the dhcpd config files. The SRM doesn't care about the name so much as it just combines them using this part of the script.

<details>
<summary>RSM <code>cat /etc/rc.network</code></summary>

<pre><code>
merge_dhcpd_conf ()
{
    local CONF_LIST=""
    local IFNAME=""
    local TAGNAME=""
    local isEnable=""
    local LEASE_FILE_NAME="dhcpd.conf.leases"
    local LEASE_FILE="dhcp-leasefile=/etc/dhcpd/${LEASE_FILE_NAME}"
    local LEASE_SCRIPT="dhcp-script=/usr/syno/etc/dhcpd/dhcpd-script.sh"
    local STATIC_HOST_CONF="/etc/dhcpd/dhcpd-static.conf"
    local RA_PARAM_CONF="/etc/dhcpd/dhcpd-raparam.conf"
    local CONF_FILE=`ls /etc/dhcpd/dhcpd-*.conf 2> /dev/null`
    local WPAD_INFO_FILE=""
    local WPAD_CONF_FILE=""
    if [ "${CONF_FILE}" = "" ]; then
        return 1
    fi

    for each in ${CONF_FILE}; do
        IFNAME=`echo ${each} | cut -d - -f 2`
        TAGNAME=`echo ${each} | cut -d - -f 3 | sed -e 's;.conf$;;'`
        isEnable=`/bin/get_key_value /etc/dhcpd/dhcpd-${IFNAME}-${TAGNAME}.info enable`
        WPAD_CONF_FILE="/etc/dhcpd/wpad-${IFNAME}.conf"

        if [ "${isEnable}" = "yes" ]; then
            CONF_LIST="${CONF_LIST} ${each}"
            if [ -f ${WPAD_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${WPAD_CONF_FILE}"
            fi
        fi

    done
    if [ "${CONF_LIST}" = "" ]; then
        return 1;
    fi
    if [ -e ${STATIC_HOST_CONF} ]; then
        CONF_LIST="${CONF_LIST} ${STATIC_HOST_CONF}"
    fi
    if [ -e ${RA_PARAM_CONF} ]; then
        CONF_LIST="${CONF_LIST} ${RA_PARAM_CONF}"
    fi

    cat ${CONF_LIST} | grep -v "${LEASE_FILE}" > ${DHCPD_CONF}
    echo ${LEASE_FILE} >> ${DHCPD_CONF}
    echo ${LEASE_SCRIPT} >> ${DHCPD_CONF}

    # if named start, disable dnsmasq DNS setting
    grep "port=0" /etc/dhcpd/dhcpd.conf
    local isPort0="$?"
    local isNamedEnable=`pidof named`
    if [ "x0" != "x${isPort0}" -a "x" != "x${isNamedEnable}" ]; then
        echo "port=0" >> /etc/dhcpd/dhcpd.conf
    fi

    return 0
}
</code></pre>
</details>

On the other-hand the DSM uses this instead.

<details>
<summary>DSM <code>cat /etc/rc.network</code></summary>
<pre><code>
merge_dhcpd_conf ()
{
        local CONF_LIST=""
        local IFNAME=""
        local TAGNAME=""
        local isEnable=""
        local LEASE_FILE_NAME="dhcpd.conf.leases"
        local LEASE_FILE="dhcp-leasefile=/etc/dhcpd/${LEASE_FILE_NAME}"
        local LEASE_SCRIPT="dhcp-script=/usr/share/dhcpd/dhcpd-script.sh"
        local WPAD_INFO_FILE=""
        local WPAD_CONF_FILE=""
        local STATIC_CONF_FILE=""
        local VENDOR_CONF_FILE=""
        local DNS_CONF_FILE=""
        local DNS_INFO_FILE=""
        local PXE_CONF_FILE=""
        local PXE_INFO_FILE=""
        local INFO_FILE=""
        local CONF_FILE=""
        local ENABLE_INTERFACE=""
        local TMP_STATIC_MERGE_FILE="/etc/dhcpd/tmp-static-merge-file"

        INFO_FILE=`/bin/ls /etc/dhcpd | /bin/grep "^dhcpd-[a-z0-9_.]*.info" 2>/dev/null`
        INFO_FILE="${INFO_FILE} `/bin/ls /etc/dhcpd | /bin/grep "^dhcpd6-[a-z0-9_.]*.info" 2>/dev/null`"
        if [ "${INFO_FILE}" = "" ]; then
                return 1
        fi

        for each in ${INFO_FILE}; do
                IFNAME=`echo ${each} | /bin/sed 's/dhcpd-//g' | /bin/sed 's/dhcpd6-//g' | /bin/sed 's/.info//g'`
                if [ "x${IFNAME}" = "xstatic" -o "x${IFNAME}" = "xdns" -o "x${IFNAME}" = "xpxe" ]; then
                        continue;
                fi

                isEnable=`/bin/get_key_value "/etc/dhcpd/"${each} enable`
                if [ "x${isEnable}" = "xyes" ]; then
                        ENABLE_INTERFACE="${ENABLE_INTERFACE} ${IFNAME}"
                fi
        done
        ENABLE_INTERFACE=`echo ${ENABLE_INTERFACE} | /bin/tr ' ' '\n' | /bin/sort -u | /bin/tr '\n' ' '`

        for IFNAME in ${ENABLE_INTERFACE}; do
                INFO_FILE=`/bin/ls /etc/dhcpd/dhcpd-${IFNAME}-subnet*.info 2>/dev/null`
                INFO_FILE="${INFO_FILE} `/bin/ls /etc/dhcpd/dhcpd6-${IFNAME}-subnet*.info 2>/dev/null`"
                IS_SLAVE="$(/bin/get_key_value /etc/sysconfig/network-scripts/ifcfg-"${IFNAME}" SLAVE)"

                if [ "yes" = "${IS_SLAVE}" ]; then
                        continue
                fi

                for each in ${INFO_FILE}; do
                        isEnable=`/bin/get_key_value ${each} enable`
                        if [ "x${isEnable}" = "xyes" ]; then
                                CONF_FILE=`echo ${each} | /bin/sed -e 's;.info$;.conf;'`
                                CONF_LIST="${CONF_LIST} ${CONF_FILE}"
                        fi
                done

                WPAD_INFO_FILE="/etc/dhcpd/wpad-${IFNAME}.info"
                WPAD_CONF_FILE="/etc/dhcpd/wpad-${IFNAME}.conf"
                isEnable=`/bin/get_key_value ${WPAD_INFO_FILE} enable`
                if [ "x${isEnable}" = "xyes" ] && [ -f ${WPAD_CONF_FILE} ]; then
                        CONF_LIST="${CONF_LIST} ${WPAD_CONF_FILE}"
                fi
        done

        if [ "${CONF_LIST}" = "" ]; then
                return 1;
        fi

        DNS_INFO_FILE="/etc/dhcpd/dhcpd-dns-dns.info"
        DNS_CONF_FILE="/etc/dhcpd/dhcpd-dns-dns.conf"
        isEnable=`/bin/get_key_value ${DNS_INFO_FILE} enable`
        if [ "x${isEnable}" = "xyes" ] && [ -f ${DNS_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${DNS_CONF_FILE}"
        fi

        PXE_INFO_FILE="/etc/dhcpd/dhcpd-pxe.info"
        PXE_CONF_FILE="/etc/dhcpd/dhcpd-pxe.conf"
        isEnable=`/bin/get_key_value ${PXE_INFO_FILE} enable`
        if [ "x${isEnable}" = "xyes" ] && [ -f ${PXE_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${PXE_CONF_FILE}"
        fi

        # compatibility for DSM4.2 ...
        STATIC_CONF_FILE="/etc/dhcpd/dhcpd-static.conf"
        if [ -f ${STATIC_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${STATIC_CONF_FILE}"
        fi

        STATIC_CONF_FILE="/etc/dhcpd/dhcpd-static-static.conf"
        if [ -f ${STATIC_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${STATIC_CONF_FILE}"
        fi

        VENDOR_CONF_FILE="/etc/dhcpd/dhcpd-vendor.conf"
        if [ -f ${VENDOR_CONF_FILE} ]; then
                CONF_LIST="${CONF_LIST} ${VENDOR_CONF_FILE}"
        fi

        cat ${CONF_LIST} | grep -v "${LEASE_FILE}" > ${DHCPD_CONF}

        /bin/rm -rf ${TMP_STATIC_MERGE_FILE}
        for IFNAME in ${ENABLE_INTERFACE}; do
                STATIC_CONF_FILE="/etc/dhcpd/dhcpd-${IFNAME}-static.conf"
                /bin/cat ${STATIC_CONF_FILE} | /bin/grep -e "dhcp-host" >> ${TMP_STATIC_MERGE_FILE}
        done
        /bin/sort -u ${TMP_STATIC_MERGE_FILE} >> ${DHCPD_CONF}
        /bin/rm -rf ${TMP_STATIC_MERGE_FILE}

        echo ${LEASE_FILE} >> ${DHCPD_CONF}
        echo ${LEASE_SCRIPT} >> ${DHCPD_CONF}

        # if named start, disable dnsmasq DNS setting
        grep "port=0" /etc/dhcpd/dhcpd.conf
        local isPort0="$?"
        local isNamedEnable=`pidof named`
        if [ "x0" != "x${isPort0}" -a "x" != "x${isNamedEnable}" ]; then
                echo "port=0" >> /etc/dhcpd/dhcpd.conf
        fi

        return 0
}
</code></pre>
</details>
I could name them both `vendor-vendor.conf` but I think having the different names makes this more future proof and doesn't mess with any other packages or user configs. See https://forum.synology.com/enu/viewtopic.php?t=131699

I also think that `dns-dns.conf` is used for the DNS server package and since I marked it incomparable with this package I think this makes it perfect for us to use.